// easy-react-text-loop/dist/cjs/TextLoop.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const framer_motion_1 = require("framer-motion");
const variants = {
    enter: (direction) => {
        return {
            y: -20,
            opacity: 0,
        };
    },
    center: {
        zIndex: 1,
        y: 0,
        opacity: 1,
    },
    exit: (direction) => {
        return {
            zIndex: 0,
            opacity: 0,
        };
    },
};
const TextLoop = ({ children, timeout = 2500, animation = "spring" }) => {
    const [index, setIndex] = (0, react_1.useState)(0);
    (0, react_1.useEffect)(() => {
        setTimeout(() => {
            const texts = react_1.default.Children.toArray(children);
            let next = index + 1;
            if (next === texts.length) {
                next = 0;
            }
            setIndex(next);
        }, timeout);
    }, [index, children, timeout]);
    return (react_1.default.createElement(framer_motion_1.AnimatePresence, null,
        react_1.default.createElement(framer_motion_1.motion.span, { style: { position: "absolute", marginLeft: "0.5rem", marginRight: "0.5rem" }, variants: variants, key: index, initial: "enter", animate: "center", exit: "exit", transition: {
                y: { type: animation, stiffness: 300, damping: 200 },
                opacity: { duration: 0.5 },
            } },
            " ",
            children && react_1.default.Children.toArray(children)[index],
            " ")));
};
exports.default = TextLoop;
